### PR TITLE
IPv4 converters

### PR DESCRIPTION
**Pull request by tshemsedinov**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [ ] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

### CODE REVIEW

File: lib/network.js [hunk 1]
```diff
@@ -42,16 +42,33 @@ const fetchPolyfill = (url, options = {}) => {
...
+const IPV4_OCTETS = 4;
+
+const ipToInt = (ip) => {
+  if (typeof ip !== 'string') return Number.NaN;
   const bytes = ip.split('.');
+  if (bytes.length !== IPV4_OCTETS) return Number.NaN;
   let res = 0;
   for (const byte of bytes) {
-    res = (res << 8) + parseInt(byte, 10);
+    res = res * 256 + parseInt(byte, 10);
```
Comment (MarhiievHE) [inline lib/network.js:53]: What do you think about leaving the shifts and adding 2^32 to res after the cycle 'for' instead of multiplying?

Comment (tshemsedinov) [reply lib/network.js:53]: The most handy format is * 256, see test cases

Comment (MarhiievHE) [reply lib/network.js:53]: My mistake, it's not just res + 2^32, it's something like this:
res = (res + Math.pow(2, 32) & MAX_32_BIT;

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/metautil/commit/55343320ceaa47dfd0f3c405059b47bff9ee1e4d, https://github.com/metarhia/metautil/commit/4eaeeb968159d80d8de6eaeb15ec5ca4fb8210aa, https://github.com/metarhia/metautil/commit/d9d7386ad85e43adcf26cf0fed58605c543bd01e, https://github.com/metarhia/metautil/commit/2bbacf3e766ebbe7aeec4e2a320f85f4236839c0, and https://github.com/metarhia/metautil/commit/f390bfa2b4d50250d0918b92274be397457066fb

File: lib/network.js (hunk 1 above)
Comment (MarhiievHE) [reply lib/network.js:53]: This means that for this to work we need to make the real sum modulo 'pow(2, 32)'
```js
res = (POW2_32 + res) % POW2_32;
```
or use the ternary operator
```js
res = res < 0 ? (POW2_32 + res) : res ;
```
but it's probably not nice.

**Pull request closed without merging**